class-gf-razorpay.php

```diff
@@ -0,0 +1,255 @@
+    protected $_url = 'http://www.gravityforms.com';
+    protected $_title = 'Gravity Forms Razorpay Add-On';
+    protected $_short_title = 'Razorpay';
+    protected $_supports_callbacks = true;
+
+    // Permissions
+    protected $_capabilities_settings_page = 'gravityforms_razorpay';
+    protected $_capabilities_form_settings = 'gravityforms_razorpay';
+    protected $_capabilities_uninstall = 'gravityforms_razorpay_uninstall';
+
+    // Automatic upgrade enabled
+    protected $_enable_rg_autoupgrade = true;
+
+    private static $_instance = null;
+
+    public static function get_instance() {
+        if ( self::$_instance == null ) {
+            self::$_instance = new GFRazorpay();
+        }
+
+        return self::$_instance;
+    }
+
+    private function __clone() {
+    }
+
+    public function init_frontend() {
+        parent::init_frontend();
+
+        add_action( 'gform_after_submission', array( $this, 'pay_using_razorpay'), 10, 2 );
+    }
+
+    public function plugin_settings_fields() {
+        return array(
+            array(
+                'title'       => '',
+                'fields'      => array(
+                    array(
+                        'name'    => 'gf_razorpay_key',
+                        'label'   => esc_html__( 'Razorpay Key', 'razorpay-gravity-forms' ),
+                        'type'    => 'text',
+                        'class'   => 'medium',
+                    ),
+                    array(
+                        'name'    => 'gf_razorpay_secret',
+                        'label'   => esc_html__( 'Razorpay Secret', 'razorpay-gravity-forms' ),
+                        'type'    => 'text',
+                        'class'   => 'medium',
+                    ),
+                    array(
+                        'type' => 'save',
+                        'messages' => array(
+                            'success' => esc_html__( 'Settings have been updated.', 'razorpay-gravity-forms' )
+                        ),
+                    ),
+                ),
+            ),
+        );
+    }
+
+    public function get_customer_fields_array($feed, $entry)
+    {
+        $fields = array();
+
+        foreach ($this->get_customer_fields() as $field)
+        {
+            $field_id = $feed['meta'][$field['meta_name']];
+
+            $value = rgar($entry, $field_id);
+
+            $fields[$field['meta_name']] = $value;
+        }
+
+        return $fields;
+    }
+
+    public function get_customer_fields() {
+        return array(
+            array( 'name' => 'first_name', 'label' => 'First Name', 'meta_name' => 'billingInformation_firstName' ),
+            array( 'name' => 'last_name', 'label' => 'Last Name', 'meta_name' => 'billingInformation_lastName' ),
+            array( 'name' => 'email', 'label' => 'Email', 'meta_name' => 'billingInformation_email' ),
+            array( 'name' => 'address1', 'label' => 'Address', 'meta_name' => 'billingInformation_address' ),
```

Comment:
@riagarg5 Right, but you're looping over them in get_customer_fields_array. Please move this to a different file. 

For eg. Please refer to the structure here: https://github.com/razorpay/razorpay-quick-payments